swiftsimio/visualisation/slice_backends/nearest_neighbors.py

```diff
@@ -0,0 +1,274 @@
+from numpy import float64, float32, ndarray, linspace, array, stack, meshgrid, power
+
+from swiftsimio import SWIFTDataset, cosmo_array
+from swiftsimio.optional_packages import KDTree, TREE_AVAILABLE
+from swiftsimio.visualisation.slice_backends.sph import get_hsml as sph_get_hsml
+
+
+def _build_tree(
+    x: float64, y: float64, z: float64, box_x: float64, box_y: float64, box_z: float64
+) -> KDTree:
+    """
+    Build the tree used for the nearest-neighbor calculations.
+    In the periodic case, we must make sure that all particle coordinates fall inside the box.
+
+    Parameters
+    ----------
+    x : array of float64
+        x-positions of the particles. Must be bounded by [0, 1].
+    y : array of float64
+        y-positions of the particles. Must be bounded by [0, 1].
+    z : array of float64
+        z-positions of the particles. Must be bounded by [0, 1].
+    box_x: float64
+        box size in x, in the same rescaled length units as x, y and z.
+        Used for periodic wrapping.
+    box_y: float64
+        box size in y, in the same rescaled length units as x, y and z.
+        Used for periodic wrapping.
+    box_z: float64
+        box size in z, in the same rescaled length units as x, y and z.
+        Used for periodic wrapping.
+
+    Returns
+    -------
+    KDTree object
+        A KD-tree built from the particle positions
+    """
+    if not TREE_AVAILABLE:
+        raise ImportError(
+            'The scipy.spatial.cKDTree class is required to use the "nearest_neighbors" slice backend.'
```

Comment:
String too long, also do not use double quotes inside as that is different to black style (feel free to change this to "The scipy.spatial.cKDTree class is required to use the 'nearest_neighbors' slice backend.")